src/Aspire.Hosting.Azure/Utils/BicepParameterNameValidator.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.RegularExpressions;
+
+namespace Aspire.Hosting.Azure.Utils;
+
+internal static partial class BicepParameterNameValidator
+{
+    [GeneratedRegex("^[a-z_]+[a-z0-9_]*$", RegexOptions.IgnoreCase)]
```

Comment:
Point to a document that explains this.

Reply:
This is related: https://github.com/Azure/bicep/issues/1494#issuecomment-776128864

Looks like bicep has some identifier's rules similar to C#.

However, I didn't find specific naming rules on bicep docs. It is only seen from src: https://github.com/Azure/bicep/blob/a992bdf2d4d7c5c7dec684b7d0de4db9cb260f8a/src/highlightjs/src/bicep.ts#L12

Reply:
I think the regex in the highlightjs implementation might be the most concise. I actually asked Open AI for a BNF for Bicep so I could figure it out, but it said that it couldn't source an authoritative document and instead inferred it from usage.

Reply:
This still needs a comment though. You can say it's from the bicep bnf